adcircpy/forcing/winds/best_track.py

```diff
@@ -232,311 +230,361 @@ def write(self, path: PathLike, overwrite: bool = False):
         with open(path, 'w') as f:
             f.write(str(self))
 
-    @property
-    def NWS(self) -> int:
-        try:
-            return self.__NWS
-        except AttributeError:
-            return 20
-
-    @NWS.setter
-    def NWS(self, NWS: int):
-        assert NWS in [8, 19, 20]
-        self.__NWS = int(NWS)
-
-    @property
-    def WTIMINC(self) -> str:
-        return (
-            f'{self.start_date:%Y %m %d %H} '
-            f'{self.df["storm_number"].iloc[0]} '
-            f'{self.BLADj} '
-            f'{self.geofactor}'
-        )
-
-    @property
-    def BLADj(self) -> float:
-        try:
-            return self.__BLADj
-        except AttributeError:
-            return 0.9
-
-    @BLADj.setter
-    def BLADj(self, BLADj: float):
-        BLADj = float(BLADj)
-        assert BLADj >= 0 and BLADj <= 1
-        self.__BLADj = BLADj
-
-    @property
-    def geofactor(self) -> float:
-        try:
-            return self.__geofactor
-        except AttributeError:
-            return 1
-
-    @geofactor.setter
-    def geofactor(self, geofactor: float):
-        geofactor = float(geofactor)
-        assert geofactor >= 0 and geofactor <= 1
-        self.__geofactor = geofactor
-
     @property
     def storm_id(self) -> str:
-        return self._storm_id
-
-    @property
-    def _storm_id(self) -> str:
-        return f'{self.basin}{self.storm_number}{self.year}'
-
-    @_storm_id.setter
-    def _storm_id(self, storm_id: str):
+        if self.__storm_id is None and not self.__invalid_storm_name:
+            if self.__dataframe is not None:
+                storm_id = get_atcf_id(
+                    storm_name=self.__dataframe['name'][0],
+                    year=self.__dataframe['datetime'][0].year,
+                )
+                try:
+                    get_atcf_file(storm_id, self.file_deck, self.mode)
+                    self.storm_id = storm_id
+                except:
+                    self.__invalid_storm_name = True
```

Comment:
if storm ID is not given, attempt to retrieve the correct storm ID from the storm name and year